File: notes/python_vs_go/loop/loop.go
package main

import "fmt"

// 只有 for 循环这一种循环语句，for 循环的这三个部分每个都可以省略
func loop1() {
	fmt.Print("loop1: ")
	for i := 0; i < 10; i++ {
		fmt.Print(i, " ")
	}
	fmt.Println()
}

func loop2() {
	fmt.Print("loop2: ")
	i := 0
	for ; i < 10; i++ {
		fmt.Print(i, " ")
	}
	fmt.Println()
}

func loop3() {
	fmt.Print("loop3: ")
	i := 0
	for ; ; i++ {
		if i == 10 {
			break
		}
		fmt.Print(i, " ")
	}
	fmt.Println()
}

func loop4() {
	fmt.Print("loop4: ")
	i := 0
	for {
		if i == 10 {
			break
		}
		fmt.Print(i, " ")
		i++
	}
	fmt.Println()
}

// for 循环也可以使用 range 表达式

func loop5() {
	fmt.Print("loop5: ")
	for i := range 10 {
		fmt.Print(i, " ")
	}
	fmt.Println()
}

func loop6() {
	fmt.Print("loop6: ")
	arr := []int{10, 11, 12, 13, 14, 15}
	for i, n := range arr {
		fmt.Printf("%d:%d ", i, n)
	}
	fmt.Println()
}

func main() {
	loop1()
	loop2()
	loop3()
	loop4()
	loop5()
	loop6()
}

File: notes/python_vs_go/loop/loop.py

words="hello world"

print("loop1: ", end="")
for w in words:
	print(w, end=' ')
print()

# for 循环使用 enumerate
print("loop2: ", end="")
for i, element in enumerate(words):
	print("%d:%s" % (i, element), end=' ')
print()

# python 可以使用 while 循环
print("loop3: ", end="")
n=10
i=0
while i<n:
	print(i, end=" ")
	i+=1
print()

# python 使用 range 函数来生成一个整型序列，这个序列的长度是可以指定的。
print("loop4: ", end="")
for i in range(10):
	print(i, end=" ")
print()
